src/GraphQL/GraphQLExtensions.cs

```diff
@@ -16,10 +16,29 @@ namespace GraphQL
     /// </summary>
     public static class GraphQLExtensions
     {
-        /// <summary>
-        /// Determines if this graph type is an introspection type.
-        /// </summary>
-        internal static bool IsIntrospectionType(this IGraphType type) => type?.Name?.StartsWith("__") ?? false;
+        private static readonly List<string> _builtInScalars = new List<string>
+        {
+            "String",
+            "Boolean",
+            "Int",
+            "Float",
+            "ID"
+        };
+
+        private static readonly List<string> _builtInDirectives = new List<string>
+        {
+            "skip",
+            "include",
+            "deprecated"
+        };
+
+        internal static bool IsIntrospectionType(this IGraphType type) => type?.Name?.StartsWith("__", StringComparison.InvariantCulture) ?? false;
+
+        internal static bool IsIntrospectionType(this string typeName) => typeName.StartsWith("__", StringComparison.InvariantCulture);
+
+        internal static bool IsBuiltInScalar(this string typeName) => _builtInScalars.Contains(typeName);
+
+        internal static bool IsBuiltInDirective(this string directiveName) => _builtInDirectives.Contains(directiveName);
```

Comment:
We should not be adding extension methods to strings that relate to type names, even for internal use.  It just doesn't make sense.  For example, this statement makes no sense:

```cs
var ret = fieldType.Description.IsIntrospectionType();
```

Let's put those in a proper class (maybe this same one) so it needs to be fully qualified such as this:

```cs
var ret = GraphQL.IsIntrospectionType(typeName);
```

You can decide where/how, and if you really just want to leave it here, that's okay -- at least it's internal.

Reply:
If these are only used by the schema printer, I might just leave them there.

Reply:
I doubted where it would be better to place these methods. As a result, I added them to `GraphQLExtensions`.

Reply:
Done